static/components/ExpectedChanges.jsx

```diff
@@ -1,51 +1,77 @@
 import * as React from 'react';
 
-import api from 'app/api';
+import LoadingIndicator from './LoadingIndicator';
+import TimeSince from './TimeSince';
 
-import TaskSummary from './TaskSummary';
+function ExpectedChanges({changes}) {
+  if (changes === undefined) {
+    return (
+      <LoadingIndicator>Fetching changes.. This may take a moment.</LoadingIndicator>
+    );
+  }
 
-function ExpectedChanges({app, env}) {
-  const [lastDeploy, setLastDeploy] = React.useState(undefined);
+  if (changes === null) {
+    return (
+      <div className="alert alert-block alert-danger">
+        There was a problem fetching the list of changes.
+      </div>
+    );
+  }
 
-  const fetchLastDeploy = React.useCallback(async () => {
-    const deploysResp = await api.request('/deploys/', {
-      query: {app, env, status: 'finished'},
-    });
+  const changeList = changes.map(commit => {
+    const resolvedCommit = commit.externalCommit ?? commit;
 
-    if (deploysResp.ok) {
-      const deploys = await deploysResp.json();
-      setLastDeploy(deploys[0] ?? null);
```

Comment:
Moved the last deploy logic into a `useLastDeploy` hook that is used in the CreateDeploy view.